lib/package.gi

```diff
@@ -1239,29 +1235,28 @@ InstallGlobalFunction( DirectoriesPackageLibrary, function( arg )
         path := arg[2];
     fi;
 
-    dirs := [];
     # We are not allowed to call
     # `InstalledPackageVersion', `TestPackageAvailability' etc.
     info:= PackageInfo( name );
     if IsBound( GAPInfo.PackagesLoaded.( name ) ) then
       # The package is already loaded.
-      version:= GAPInfo.PackagesLoaded.( name )[2];
+      ppath:= GAPInfo.PackagesLoaded.( name )[1];
     elif IsBound( GAPInfo.PackageCurrent ) then
       # The package is currently going to be loaded.
-      version:= GAPInfo.PackageCurrent.Version;
+      ppath:= GAPInfo.PackageCurrent.InstallationPath;
     elif 0 < Length( info ) then
-      # Take the installed package with the highest version.
-      version:= info[1].Version;
-    fi;
-    for r in info do
-      if r.Version = version then
-        tmp:= Concatenation( r.InstallationPath, "/", path );
-        if IsDirectoryPath( tmp ) = true then
-          Add( dirs, Directory( tmp ) );
-        fi;
-      fi;
-    od;
-    return dirs;
+      # Take the installed package with the highest version
+      # that has been found first in the root paths.
+      ppath:= info[1].InstallationPath;
+    else
+      # This package is not known.
+      return [];
+    fi;
+    tmp:= Concatenation( ppath, "/", path );
+    if IsDirectoryPath( tmp ) = true then
```

Comment:
I notice that we check `IsDirectoryPath` here, but not in `DirectoriesPackagePrograms`... I wonder why?

Also, perhaps the two functions could share some code in a common helper (I am not meaning to request that you do this in this PR, just thinking out loud).

Reply:
Yes, `DirectoriesPackageLibrary` checks the path, and `DirectoriesPackagePrograms` doesn't.
Users may rely on this behaviour, and I had no reason to change this.